sdk/messaging/azservicebus/receiver.go

```diff
@@ -196,7 +197,8 @@ func (r *Receiver) ReceiveMessages(ctx context.Context, maxMessages int, options
 		return nil, errors.New("receiver is already receiving messages. ReceiveMessages() cannot be called concurrently")
 	}
 
-	return r.receiveMessagesImpl(ctx, maxMessages, options)
+	messages, err := r.receiveMessagesImpl(ctx, maxMessages, options)
+	return messages, internal.TransformError(err)
```

Comment:
Is it possible to move this error transformation further down the stack so that implementers of new APIs don't need to remember to do this?

Reply:
If you want to save that refactoring for post-GA that's fine, just open an issue to track it.

Reply:
This approach was a bit annoying but it has the benefit of letting me shuffle the internals around more without having to worry about which piece of code was responsible for transforming it.

It's not a huge deal, but this was a bit easier to manually review.